filters/utilization_threshold.cpp

```diff
@@ -0,0 +1,105 @@
+#include <memory>
+#include <string>
+
+#include "glog/logging.h"
+
+#include "filters/utilization_threshold.hpp"
+
+#include "mesos/resources.hpp"
+
+#include "serenity/metrics_helper.hpp"
+
+namespace mesos {
+namespace serenity {
+
+using std::string;
+
+Try<Nothing> UtilizationThresholdFilter::consume(const ResourceUsage& product) {
+  std::unique_ptr<ExecutorSet> newSamples(new ExecutorSet());
+  double_t totalCpuUsage = 0;
+
+  for (const ResourceUsage_Executor& inExec : product.executors()) {
+    // In case of lack of the usage for given executor or lack of
+    // executor_info filter assumes that it uses maximum of allowed
+    // resource (allocated).
+    bool useAllocatedForUtilization = false;
+    string executor_id = "<unknown>";
+
+    if (!inExec.has_executor_info()) {
+      LOG(WARNING) << "Executor " << executor_id
+      << " does not include executor_info. "
+      << UTILIZATION_THRESHOLD_FILTER_WARNING;
+      // We can work without that - using allocated.
+      useAllocatedForUtilization = true;
+    } else {
+      executor_id = inExec.executor_info().executor_id().value();
+    }
+
+    if (!inExec.has_statistics()) {
+      LOG(WARNING) << "Executor " << executor_id
+      << " does not include statistics. "
+      << UTILIZATION_THRESHOLD_FILTER_WARNING;
+      // We can work without that - using allocated resources.
+      useAllocatedForUtilization = true;
+    }
+
+    if (inExec.allocated().size() == 0) {
```

Comment:
In theory there is such method: https://developers.google.com/protocol-buffers/docs/reference/cpp/google.protobuf.repeated_field#RepeatedPtrField However our compiler does not allow it. ):
![image](https://cloud.githubusercontent.com/assets/6950331/8697796/5963c62c-2af8-11e5-90be-e4997afc1bce.png)